superset/daos/base.py

```diff
@@ -179,26 +183,33 @@ def update(cls, model: T, properties: dict[str, Any], commit: bool = True) -> T:
             raise DAOUpdateFailedError(exception=ex) from ex
         return model
 
+    @overload
+    @classmethod
+    def delete(cls, objects: list[T], commit: bool = True) -> None:
+        ...
+
+    @overload
+    @classmethod
+    def delete(cls, objects: T, commit: bool = True) -> None:
+        ...
+
     @classmethod
-    def delete(cls, model: T, commit: bool = True) -> T:
+    def delete(cls, objects: T | list[T], commit: bool = True) -> None:
```

Comment:
`objects` seems to be a more accurate than `models`, i.e., we're deleting database objects as opposed to models.